pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -625,23 +621,30 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 			WithField("syncAPIVersion", resourceSyncStatus.APIVersion).
 			WithField("syncKind", resourceSyncStatus.Kind)
 
+		info, discoveryErr = h.Info(resource.Raw)
+		// If an error occurred, stop processing right here
+		if discoveryErr != nil {
+			resourceSyncStatus.Conditions = r.setUnknownObjectSyncCondition(resourceSyncStatus.Conditions, discoveryErr)
```

Comment:
We should check with our users to make sure that nobody is reading the UnknownObjectSync condition from its current location.

Reply:
@igoihman We are now putting the `UnknownObjectSync` condition on each individual resource instead of the syncset. Are we ok with it from your end?

Reply:
we are not checking this condition - it's ok from my end.